omniscidb/QueryEngine/QueryMemoryInitializer.cpp

```diff
@@ -383,6 +383,12 @@ QueryMemoryInitializer::QueryMemoryInitializer(
     group_by_buffers_.push_back(varlen_output_buffer);
   }
 
+  ColBuffersPtr res_set_col_buf = std::make_shared<const ColBuffers>(col_buffers);
+  FragOffsetsPtr res_set_col_frag_offsets = std::make_shared<const FragOffsets>(
+      get_col_frag_offsets(ra_exe_unit.target_exprs, frag_offsets));
+  FragSizesPtr res_set_frag_sizes = std::make_shared<const FragSizes>(
+      get_consistent_frags_sizes(ra_exe_unit.target_exprs, consistent_frag_sizes));
```

Comment:
BTW I suspect that these calls moved out of the loop can be more important for perf than vector copies.

Reply:
It didn't help much, `shared_ptr` was what brought things to ~1ms